text/0000-guaranteed-tco.md

```diff
@@ -0,0 +1,608 @@
+- Feature Name: guaranteed_tco
+- Start Date: 2023-04-01
+- RFC PR: [rust-lang/rfcs#0000](https://github.com/rust-lang/rfcs/pull/0000)
+- Rust Issue: [rust-lang/rust#0000](https://github.com/rust-lang/rust/issues/0000)
+
+# Summary
+[summary]: #summary
+
+This feature provides a guarantee that function calls are tail-call optimized via the `become` keyword. If this
+guarantee can not be provided by the compiler an error is generated instead.
```

Comment:
```suggestion
This feature provides a guarantee that function calls are tail-call optimized via the `become` keyword. If this
guarantee can not be provided by the compiler an error is generated by the compiler instead.
```

Reply:
Just want to make sure that it is clear to everyone the error is not generated at program runtime but at compile time.

Reply:
Good point, though I think this wording would be even clearer:
> ... can not be provided by the compiler a compile time error is generated instead.

Reply:
yes that is even better :)